File: plugin/com.hello.uxml.tools.ui/src/com/hello/uxml/tools/framework/graphics/SolidBrush.java
package com.hello.uxml.tools.framework.graphics;

/**
 * Represents a Brush with a single solid color.
 *
 * @author ferhat
 */
public class SolidBrush extends Brush {

  /** brush color */
  private Color color;

  /**
   * Constructor.
   */
  public SolidBrush() {
    this.color = Color.EMPTY;
  }

  /**
   * Constructor.
   */
  public SolidBrush(Color color) {
    this.color = color;
  }

  /**
   * Returns brush color.
   */
  public Color getColor() {
    return color;
  }

  public void setColor(Color value) {
    color = value;
  }
}

File: framework/lib/solid_brush.dart
part of uxml;

/**
 * Represents a Brush with a single solid color.
 *
 * @author ferhat@ (Ferhat Buyukkokten)
 */
class SolidBrush extends Brush {
  /** brush color */
  Color _color;

  /**
   * Constructor.
   */
  SolidBrush(Color color) : super() {
    _type = 0;
    _color = color;
  }

  /**
   * Sets/returns brush color.
   */
  Color get color {
    return _color;
  }

  set color(Color value) {
    _color = value;
  }

  /**
   * Creates a solid brush from rgb value.
   * TODO(ferhat): deprecate, change to dart ctor style.
   */
  static SolidBrush fromRGB(int rgb) {
    return new SolidBrush(Color.fromRGB(rgb));
  }

  /**
   * Clone brush.
   */
  Brush clone() {
    return new SolidBrush(new Color(_color.argb));
  }
}
